src/auth/dto/sign-up-response.ts

```diff
@@ -1,9 +1,19 @@
 import { User } from '@prisma/client';
 
 export class SignUpResponse {
-  constructor(public user: User) {}
+  public id: number;
+  public email: string;
+  public password: string;
+  public role: string;
+
+  constructor(id: number, email: string, password: string, role: string) {
+    this.id = id;
+    this.email = email;
+    this.password = password;
+    this.role = role;
+  }
```

Comment:
I guess you are doing it this way so that the DTOs are properly displayed in the Swagger? Have you searched for a solution to have Swagger read DTOs correctly and define the fields only in `constructor`?

Reply:
Yes I've searched for solution to this problem, but I couldn't find anything what can solve that.

Reply:
ok, I guess we will leave it like that then. I guess we would probably have to define it like this either way if we want to use `@ ApiProperty` (not sure about this though)